cumulus_etl/common.py

```diff
@@ -32,49 +31,13 @@ def ls_resources(root, resource: str) -> list[str]:
 #
 ###############################################################################
 
-_user_fs_options = {}  # don't access this directly, use get_fs_options()
 
-
-def set_user_fs_options(args: dict) -> None:
-    """Records user arguments that can affect filesystem options (like s3_region)"""
-    _user_fs_options.update(args)
-
-
-def get_fs_options(protocol: str) -> dict:
-    """Provides a set of storage option kwargs for fsspec calls or pandas storage_options arguments"""
-    options = {}
-
-    if protocol == "s3":
-        # Check for region manually. If you aren't using us-east-1, you usually need to specify the region
-        # explicitly, and fsspec doesn't seem to check the environment variables for us, nor pull it from
-        # ~/.aws/config
-        region_name = _user_fs_options.get("s3_region")
-        if region_name:
-            options["client_kwargs"] = {"region_name": region_name}
-
-        # Assume KMS encryption for now - we can make this tunable to AES256 if folks have a need.
-        # But in general, I believe we want to enforce server side encryption when possible, KMS or not.
-        options["s3_additional_kwargs"] = {
-            "ServerSideEncryption": "aws:kms",
-        }
-
-        # Buckets can be set up to require a specific KMS key ID, so allow specifying it here
-        kms_key = _user_fs_options.get("s3_kms_key")
-        if kms_key:
-            options["s3_additional_kwargs"]["SSEKMSKeyId"] = kms_key
-
-    return options
-
-
-def open_file(path: str, mode: str):
-    """A version of open() that handles remote access, like to S3"""
-    # Grab protocol if present
-    parsed = urlparse(path)
-    protocol = parsed.scheme or "file"  # assume local if no obvious scheme
-
-    # We pass auto_mkdir because on some backends (like S3), we may not have permissions that fsspec might want,
-    # like CreateBucket. We elsewhere call Root.makedirs as needed.
-    return fsspec.open(path, mode, encoding="utf8", auto_mkdir=False, **get_fs_options(protocol))
+@contextlib.contextmanager
+def open_file(path: str, mode: str) -> TextIO:
+    """A version of open() that handles atomic file access across many remote filesystems (like S3)"""
```

Comment:
- how would you feel about calling this `atomic_open_file` or `open_remote_file`?
- re the docstring, i believe you, but it's not apparent from me looking at the function why this provides better atomicity.works.

Reply:
> how would you feel about calling this atomic_open_file or open_remote_file?

Hmm, `open_remote_file` is misleading a bit, because it also handles local atomicity. I know the docstring calls out remote, but that's because that's harder and more impressive to offer. :smile: 

I can remove the word `remote` there though to avoid confusion.

As for `atomic_open_file` -- I'm a little resistant but could be convinced. (A) I'd probs want to mark it with an underscore then because that's a bit of a mouthful for a public method. (B) emphasizing the atomicity like that was not my first instinct because really, this is just fixing `open_file` to work like the ETL wants (and most sane high-level-I/O folks would expect) -- we never want non-atomic writes in any part of our ETL, this is just how the ETL _does_ I/O, so there's not like two versions of this, one with an atomic prefix and one without.

> re the docstring, i believe you, but it's not apparent from me looking at the function why this provides better atomicity/works.

I can add a comment next to the transaction line - but `fsspec` by default is not atomic - it will write out what it has done so far upon an error. So the `with transaction` line in this method is how fsspec offers atomicity - it will no longer commit the file upon an error, but instead abandon it.

(For local filesystems, it writes to temp and moves the file into place when done. For S3, it always uploads files as non-visible multipart chunks anyway, and either commits them to the final place on success or abandons the chunks on failure. Without a transaction in place, fsspec commits by default even on failure)

Reply:
Cleaned up docstring and added a comment near the transaction line

Reply:
Hmm, actually the ETL does sometimes do partial writes - to local ndjson log/download files  -- but that's already abstracted separately.

Reply:
Hmm, maybe I'm being resistant for no reason - I'll rename it `_atomic_open()`